src/handler/mod.rs

```diff
@@ -1152,9 +1161,14 @@ impl Handler {
         self.active_requests.insert(node_address, request_call);
     }
 
-    fn new_session(&mut self, node_address: NodeAddress, session: Session) {
+    async fn new_session<P: ProtocolIdentity>(
+        &mut self,
+        node_address: NodeAddress,
+        session: Session,
+    ) {
         if let Some(current_session) = self.sessions.get_mut(&node_address) {
             current_session.update(session);
+            self.replay_active_requests::<P>(&node_address).await;
```

Comment:
changes like this deserve a comment.
I need to think this one a bit more but the session stores the old keys in case they are needed, so I'm not sure this is necessary

Reply:
Added a comment, but lmk if you feel like this needs to be removed. Although just to mention that this was the change that solved the `FindNode` requests timeout error described above, but if this is unnecessary I'll debug that failure some more and find a more appropriate solution

Reply:
[The timeout](https://github.com/sigp/discv5/pull/200#issuecomment-1657275126) has been solved by this change however I noticed another issue about this replaying:

As highlighted by the red square below, `replay_active_requests()` re-sends the `FINDNODE(id:1)` that has already been sent in the Handshake message. Active requests that should be re-sent is only `FINDNODE(id:2)` in the case of this example.

```mermaid
sequenceDiagram
    participant Node1
    participant Node2
    Note over Node1: Start discv5 server
    Note over Node2: Start discv5 server

    Node1 ->> Node2: message
    Node2 ->> Node1: message
    Note over Node1,Node2: Session established

    rect rgb(100, 100, 0)
    Note over Node2: ** Session expired **
    end

    rect rgb(10, 10, 10)
    Note left of Node1: Node1 sends two requests<br> **in parallel**.
    par 
    Node1 ->> Node2: FINDNODE (id:1)
    and 
    Node1 -->> Node2: FINDNODE (id:2)
    end
    end

    Note over Node2: Send WHOAREYOU<br>since the session has been expired
    Node2 ->> Node1: WHOAREYOU (id:1)

    rect rgb(100, 100, 0)
    Note over Node2: Drop FINDNODE(id:2) request<br>since WHOAREYOU already sent.
    end

    Note over Node1: New session established with Node2

    Node1 ->> Node2: Handshake message (id:1)

    Note over Node2: New session established with Node1

    rect rgb(10, 10, 10)
    Note left of Node1: Handler::replay_active_requests()
    rect rgb(100, 0, 0)
    Node1 ->> Node2: FINDNODE (id:1) <br>**duplicates with Handshake message (id:1)**
    end
    Node1 -->> Node2: FINDNODE (id:2)
    end

    Node2 ->> Node1: NODES (id:1)
    Node2 -->> Node1: NODES (id:2)

    rect rgb(100, 0, 0)
    Node2 ->> Node1: NODES (id:1)
    end
```

Reply:
Cool, thanks for the diagram it was super helpful. I believe I've implemented a sufficient solution that filters out the handshake request when replaying active requests. But only in the case when `Node2` sends the `whoareyou`. I'm not sure if I took the right approach here, but lmk if you see anything that needs changing.